lib/couchx/constraint.ex

```diff
@@ -7,21 +7,27 @@ defmodule Couchx.Constraint do
       fields,
       prev_fields
     ) do
-    params = Enum.into(fields, %{})
-    changeset = schema.changeset(schema.__struct__, params)
-
-    fields = Keyword.merge(prev_fields, fields)
-
-    changeset
-    |> Map.get(:constraints)
-    |> Enum.map(&process_constraints(&1, source, fields, server, prev_fields))
+      if with_schema?(schema) do
+        params = Enum.into(fields, %{})
+        changeset = schema.changeset(schema.__struct__, params)
+        fields = Keyword.merge(prev_fields, fields)
+
+        changeset
+        |> Map.get(:constraints)
```

Comment:
What will `constraints` return? The uniqueness validation?

Reply:
there are 2 kind of constraints, uniqueness and foreign_key_existence, depending on the constraints used, the changeset will have them... you can try this by setting a constraint in the change set and call the `constraints` key on the changeset struct